thrust/testing/merge_by_key.cu

```diff
@@ -120,47 +102,68 @@ void TestMergeByKeyDispatchImplicit()
 
   ASSERT_EQUAL(13, vec.front());
 }
+
+template <typename T, typename CompareOp, typename... Args>
+auto call_merge_by_key(Args&&... args) -> decltype(thrust::merge_by_key(std::forward<Args>(args)...))
+{
+  _CCCL_IF_CONSTEXPR (::cuda::std::is_void<CompareOp>::value)
+  {
+    return thrust::merge_by_key(std::forward<Args>(args)...);
+  }
+  else
+  {
+    // TODO(bgruber): remove next line in C++17 and pass CompareOp{} directly to stable_sort
+    using C = ::cuda::std::__conditional_t<::cuda::std::is_void<CompareOp>::value, thrust::less<T>, CompareOp>;
+    return thrust::merge_by_key(std::forward<Args>(args)..., C{});
+  }
```

Comment:
This is a bit annoying to handle, but I want to test calling the thrust API with and without a comparison functor. C++11 makes this additionally hard, since it wants to compile both branches of the `if constexpr`.